cmd/rollkit/commands/intercept.go

```diff
@@ -0,0 +1,125 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	cometos "github.com/cometbft/cometbft/libs/os"
+	"github.com/spf13/cobra"
+
+	rollconf "github.com/rollkit/rollkit/config"
+)
+
+const rollupBinEntrypoint = "entrypoint"
+
+var (
+	rollkitConfig rollconf.TomlConfig
+
+	// ErrRollkitCommand is returned when the command is a rollkit command
+	// which we don't want to intercept.
+	ErrRollkitCommand = fmt.Errorf("rollkit command")
+
+	// ErrHelpVersion is returned when the user attempts to run help or version.
+	ErrHelpVersion = fmt.Errorf("help or version")
+
+	// ErrRunEntrypoint is returned when the entrypoint fails to run.
+	ErrRunEntrypoint = fmt.Errorf("run rollup entrypoint")
+)
+
+// InterceptCommand intercepts the command and runs it against the `entrypoint`
+// specified in the rollkit.toml configuration file.
+func InterceptCommand(
+	rollkitCommand *cobra.Command,
+	readToml func() (rollconf.TomlConfig, error),
+	runEntrypoint func(*rollconf.TomlConfig, []string) error,
+) error {
+	skipCommands := []string{}
+	for _, cmd := range rollkitCommand.Commands() {
+		skipCommands = append(skipCommands, cmd.Use)
+	}
+
+	if len(os.Args) >= 2 {
+		// check if user attempted to run a rollkit command
+		// if so, we don't want to intercept it except for `start`
+		for _, skipCmd := range skipCommands {
+			if os.Args[1] == skipCmd && os.Args[1] != "start" {
+				return ErrRollkitCommand
+			}
+		}
+
+		// check if user attempted to run help or version
+		switch os.Args[1] {
+		case "help", "--help", "h", "-h", "version", "--version", "v", "-v":
+			return ErrHelpVersion
+		}
+	}
+
+	var err error
+	rollkitConfig, err = readToml()
+	if err != nil {
+		return err
+	}
+
+	if rollkitConfig.Entrypoint == "" {
+		return fmt.Errorf("no entrypoint specified in %s", rollconf.RollkitToml)
+	}
+
+	flags := []string{}
+	if len(os.Args) >= 2 {
+		flags = os.Args[1:]
+	}
+
+	return runEntrypoint(&rollkitConfig, flags)
```

Comment:
Building on previous comment and cleaning up a few things in this method.

- I think the complexity of this operation requires 2 return values to more cleanly handle different cases. 
- there are 2 `if len(os.Args) >= 2` conditions, we can combine
- there isn't a reason to build the `skipCommands` slice. just use that for loop

```suggestion
func InterceptCommand(
	rollkitCommand *cobra.Command,
	readToml func() (rollconf.TomlConfig, error),
	runEntrypoint func(*rollconf.TomlConfig, []string) error,
) (bool, error) {
        // Grab flags and verify command
	flags := []string{}
	if len(os.Args) >= 2 {
		flags = os.Args[1:]
		
		// Handle specific cases first for help, version, and start
		switch os.Args[1] {
		case "help", "--help", "h", "-h":
		fallthrough
		case "version", "--version", "v", "-v":
			return false, nil
		case "start":
			goto readTOML
		}
		
		// check if user attempted to run a rollkit command
		for _, skipCmd := range rollkitCommand.Commands() {
			if os.Args[1] == skipCmd.Use {
				return false, nil
			}
		}
	}
readTOML:
	var err error
	rollkitConfig, err = readToml()
	if err != nil {
		return false, err
	}

        // After successfully reading the TOML file, we expect to be able to use the entrypoint
	if rollkitConfig.Entrypoint == "" {
		return true, fmt.Errorf("no entrypoint specified in %s", rollconf.RollkitToml)
	}

	return true, runEntrypoint(&rollkitConfig, flags)
```